subprojects/core-api/src/main/java/org/gradle/api/tasks/TaskDestroyables.java

```diff
@@ -27,23 +26,31 @@
 @Incubating
 public interface TaskDestroyables {
     /**
+     * Registers files or directories that this task destroys.
+     *
+     * @param paths The files or directories that will be destroyed. The given paths are evaluated as per {@link org.gradle.api.Project#files(Object...)}.
+     *
+     * @since 4.3
+     */
+    void register(Object... paths);
+
+    /**
      * Registers some files that this task destroys.
      *
      * @param paths The files that will be destroyed. The given paths are evaluated as per {@link org.gradle.api.Project#files(Object...)}.
+     *
+     * @deprecated Use {@link #register(Object...)} instead.
```

Comment:
Since this class is incubating, I wonder if we shouldn't just remove the `file()`/`files()` methods.  I don't see a strong argument for keeping them around in deprecated form - anyone using these methods has already accepted the implications of `@Incubating` and changing it is just a matter changing the method being called.

Reply:
I don't mind adding the deprecation: it helps people support more than a single version of Gradle with whatever plugin they might have. I don't feel very strongly about this, though. If you think we should not have these methods anymore, I'm happy to remove them.

Reply:
I think it's a fair point, I'm just trying to avoid having to clean this up later.  I wonder how much meaning `@Incubating` has, though, if we deprecate rather than remove even incubating things.